pkg/values/value.go

```diff
@@ -303,12 +302,18 @@ func fromBigIntValueProto(biv *pb.BigInt) *BigInt {
 }
 
 func CreateMapFromStruct(v any) (*Map, error) {
-	var resultMap map[string]interface{}
-
-	err := mapstructure.Decode(v, &resultMap)
-	if err != nil {
-		return nil, err
+	resultMap := map[string]any{}
+
+	// use reflect to handle nested types as an interface
+	rv := reflect.ValueOf(v)
+	rt := reflect.TypeOf(v)
+	for i := 0; i < rv.NumField(); i++ {
+		if !rt.Field(i).IsExported() {
```

Comment:
@justinkaseman We can just skip unexported fields, i.e. treat them as private. This is pretty standard eg. across json.Marshal/Unmarshal, proto.Marshal/Unmarshal etc.